File: core/src/com/xenoage/zong/core/music/util/FirstOrLast.java
package com.xenoage.zong.core.music.util;

import com.xenoage.zong.core.music.VoiceElement;


/**
 * The side where to search for a {@link VoiceElement},
 * i.e. either the first matching element or the last one.
 * 
 * @author Andreas Wenger
 */
public enum FirstOrLast {
	/** The first element (at the lowest beat) that matches a condition. */
	First,
	/** The last element (at the highest beat) that matches a condition. */
	Last;
}

File: core/src/com/xenoage/zong/core/music/util/FirstOrLast.kt
package com.xenoage.zong.core.music.util

import com.xenoage.zong.core.music.VoiceElement


/**
 * The side where to search for a [VoiceElement],
 * i.e. either the first matching element or the last one.
 */
enum class FirstOrLast {
	/** The first element (at the lowest beat) that matches a condition. */
	First,
	/** The last element (at the highest beat) that matches a condition. */
	Last
}
